File: patterns-modules/design-patterns-behavioral/src/main/java/com/baeldung/visitor/Document.java
package com.baeldung.visitor;

import java.util.ArrayList;
import java.util.List;

public class Document extends Element {

    List<Element> elements = new ArrayList<>();

    public Document(String uuid) {
        super(uuid);
    }

    @Override
    public void accept(Visitor v) {
        for (Element e : this.elements) {
            e.accept(v);
        }
    }
}

File: core-kotlin-2/src/main/kotlin/com/baeldung/jvmannotations/Document.kt
package com.baeldung.jvmannotations

import java.util.*

interface Document {

    @JvmDefault
    fun getType() = "document"
}

class TextDocument : Document {
    override fun getType() = "text"

    fun transformList(list : List<Number>) : List<Number> {
        return list.filter { n -> n.toInt() > 1 }
    }

    fun transformListInverseWildcards(list : List<@JvmSuppressWildcards Number>) : List<@JvmWildcard Number> {
        return list.filter { n -> n.toInt() > 1 }
    }

    var list : List<@JvmWildcard Any> = ArrayList()
}

class XmlDocument(d : Document) : Document by d

fun main() {
    val myDocument = TextDocument()
    val myTextDocument = XmlDocument(myDocument)
    println("${myDocument.getType()} ${myTextDocument.getType()}")
}
